lib/auth/auth_with_roles.go

```diff
@@ -1666,17 +1666,41 @@ func (a *ServerWithRoles) GetToken(ctx context.Context, token string) (types.Pro
 	return a.authServer.GetToken(ctx, token)
 }
 
+func enforceEnterpriseJoinMethodCreation(token types.ProvisionToken) error {
+	if modules.GetModules().BuildType() == modules.BuildEnterprise {
+		return nil
+	}
+
+	v, ok := token.(*types.ProvisionTokenV2)
+	if !ok {
+		return trace.BadParameter("unexpected token type %T", token)
+	}
+
+	if v.Spec.GitHub != nil && v.Spec.GitHub.EnterpriseServerHost != "" {
+		return trace.AccessDenied(
+			"github enterprise server joining support requires enterprise license",
+		)
+	}
```

Comment:
In below code the  `EnterpriceServerHost` field is marked as a optional field allowing to overwrite `github.com` hostname:  [link](https://github.com/gravitational/teleport/pull/19644/files#diff-47dae730c9f24f1dccd0bcae2819f8c210b58bcab1404b8fe29a8f9f0cca260cR50-R53) where the check `modules.BuildEnterprise` is done at the beginning of this function. 

How it is possible that a client has Enterprise licence but it can't use enterprise feature ?

Reply:
Is this code not enforcing that an enterprise license is needed to set this field ?